packages/hooks/src/useTransferBetweenAccounts.ts

```diff
@@ -69,79 +71,72 @@ const useTransferBetweenAccounts = () => {
                                 )?.market_type,
                             }),
                         };
-                    }) || [],
+
+                        return all_accounts;
+                    },
+                    {} as Record<
+                        string,
+                        NonNullable<
+                            typeof accounts[number] & {
+                                gradient_class?: `wallet-card__${string}`;
+                                icon?: string;
+                                mt5_market_type?: 'all' | 'financial' | 'synthetic';
+                            }
+                        >
+                    >
+                ) || {},
             wallets:
-                accounts
-                    ?.filter(account => account.account_type === 'wallet')
-                    .map(wallet_account => {
-                        const wallet = all_wallets?.find(acc => acc.loginid === wallet_account.loginid);
-
-                        return {
-                            ...wallet_account,
-                            icon: wallet?.icon,
-                            gradient_class: wallet?.gradient_card_class,
+                accounts?.reduce(
```

Comment:
Instead of transforming it here just for useTransferBetweenAccounts. Can you move it to a generic hook? Like useAccounts which gets the data and transform it in a central hook? Other people might need to use this in the future